moha/rauk/rauk.py

```diff
@@ -98,36 +119,36 @@ def assign_rauk_parameters(
         # Paths to the JSON files
         hx_dictionary_path = Path(__file__).parent / "hx_dictionary.json"
         hx_dictionary = json.load(open(hx_dictionary_path, "rb"))
-
         kxy_matrix_1_path = Path(__file__).parent / "kxy_matrix_1.json"
         kxy_matrix_1_list = json.load(open(kxy_matrix_1_path, "rb"))
-
         # Convert list back to numpy array
         kxy_matrix_1 = np.array(kxy_matrix_1_list)
         kxy_matrix = np.minimum(
             kxy_matrix_1,
             kxy_matrix_1.T)  # Symmetric matrix
-
         alpha_c = -0.414  # Value for sp2 orbital of Carbon atom.
         beta_c = -0.0533  # Value for sp2 orbitals of Carbon atom.
-
         # Create bond dictionary using predefined values without overlap
         # parameters
         bond_dictionary = {}
-        for i, atom in enumerate(atom_types):
-            next_atom = atom_types[i + 1]\
-                if i < len(atom_types) - 1 else atom_types[0]
-            index1 = list(hx_dictionary.keys()).index(atom)
-            index2 = list(hx_dictionary.keys()).index(next_atom)
-            bond_key = atom + next_atom
+
+        for atom1, atom2, _ in connectivity:
+            atom1_name, site1 = get_atom_type(atom1)
+            atom2_name, site2 = get_atom_type(atom2)
+
+            # get indices of atoms
+            index1 = list(hx_dictionary.keys()).index(atom1_name)
+            index2 = list(hx_dictionary.keys()).index(atom2_name)
+
+            bond_key = ','.join([atom1_name, atom2_name])
             bond_value = kxy_matrix[index1, index2] * abs(beta_c)
             bond_dictionary[bond_key] = bond_value
-            bond_dictionary[next_atom + atom] = bond_value  # Ensure symmetry
+            bond_dictionary[','.join([atom2_name, atom1_name])] = bond_value
+            # Ensure symmetry
```

Comment:
We should swap lines 146 and 147, to make sure that comment precedes the line it's related to